bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -36,7 +36,10 @@ async fn download_snapshot_header(download_url: &str) -> Result<SnapshotHeader,
 
                     let mut slice: &[u8] = &bytes[..SnapshotHeader::LENGTH];
 
-                    return Ok(SnapshotHeader::unpack(&mut slice)?);
+                    return Ok(SnapshotHeader::unpack_verified(&mut slice).map_err(|err| match err {
+                        UnpackError::Packable(err) => err,
+                        UnpackError::Unpacker(_) => unreachable!(),
```

Comment:
The issue here is that we are assuming that the `slice` has enough bytes to unpack a `SnapshotHeader` because the size of it is constant.

Reply:
well the thing is that it cannot error because `SnapshotHeader::LEN` should be equal to `SnapshotHeader::packed_len` all the time. Which means that if we reach this code path we messed up `SnapshotHeader` or its unpacking.

Reply:
Ahhh I see, thanks

Reply:
Maybe a tiny little comment, took me time to understand 😬